plugin/build.gradle.kts

```diff
@@ -69,3 +107,7 @@ tasks.jacocoTestReport {
 tasks.check {
     dependsOn(tasks.jacocoTestReport)
 }
+
+tasks.withType<Test> {
+    dependsOn("publishDefaultPublicationToPluginTestRepository")
```

Comment:
This is the main trick to test the plugin. It'll deploy the plugin artifacts to `rootDir/build/localMaven` so that the testProject can use this one and not the one on the gadle portal.